tests/test_client.py

```diff
@@ -196,3 +197,26 @@ def test_client_credentials_grant(cc_grant_tuple):
     assert gc.auth_client.oauth2_validate_token(gc.garden_authorizer.access_token)[
         "active"
     ]
+
+
+def test_register_pipeline_with_specified_uuid(
+    mocker, garden_client, pipeline_toy_example
+):
+    CONTAINER_UUID = "3dc3170e-2cdc-4379-885d-435a0d85b581"
+    pipeline_toy_example.container_uuid = CONTAINER_UUID
+
+    # Mock register_pipeline so we can see if the container uuid gets passed in
+    mocker.patch(
+        "garden_ai.client.register_pipeline",
+        return_value="a4742a1b-966a-4913-a369-fb6555b8bfb9",
```

Comment:
not that we use it, but wouldn't the return value here now be a `RegisteredPipeline`?

Reply:
Confusingly, this is mlmodel.register_pipeline being mocked. Just mocking it at the import site. mlmodel.register_pipeline does indeed return a uuid